### PR TITLE
Replaces icon and buttons styles with gosheets classes

### PR DESCRIPTION
**Pull request by AlexOverbeck**
https://mobi.teamwork.com/#tasks/12980128

This refactors the buttons and icon components to accept the BEM
style classes defined in the gosheets module. It also fixes a bug
with the button component where the buttonVariant was being applied
to the buttons type attribute.

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-button/go-button.component.ts [hunk 1]
```diff
@@ -18,22 +17,28 @@ export class GoButtonComponent {
 
   constructor() {}
 
-  public clicked() : void {
-    if (this.isProcessing || this.buttonDisabled) { return; }
```
Comment (AlexOverbeck) [inline projects/go-lib/src/lib/components/go-button/go-button.component.ts]: We don't actually need this protection if we utilize the disabled attribute properly.

File: projects/go-lib/src/lib/components/go-icon/go-icon.component.ts [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
-  templateUrl: 'go-icon.component.html',
-  styleUrls: ['go-icon.component.scss']
+  templateUrl: 'go-icon.component.html'
 })
 export class GoIconComponent {
   @Input() icon: string;
+  @Input() iconModifier: string;
+  @Input() iconClass: string;
+
+  public classObject(): object {
```
Comment (AlexOverbeck) [inline projects/go-lib/src/lib/components/go-icon/go-icon.component.ts]: We should probably think of a consistent, potentially more performant way to handle this sort of class manipulation.

File: projects/go-lib/src/lib/components/go-button/go-button.component.ts [hunk 3]
```diff
@@ -18,22 +17,28 @@ export class GoButtonComponent {
...
   
-  public reset() : void {
+  public reset(): void {
     this.isProcessing = false;
   }
 
-  public classList() : any {
+  public classObject(): object {
+    // 'alert' as a variant is depreciated and
+    // will be removed in a later version
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-button/go-button.component.ts]: I like this concept, allows us to communicate deprecated changes ahead of time. :+1:

File: projects/go-lib/src/lib/components/go-icon/go-icon.component.html [hunk 4]
```diff
@@ -1 +1,6 @@
-<i class="material-icons">{{icon}}</i>
+<i
+  class="go-icon"
+  [ngClass]="classObject()"
+>
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-icon/go-icon.component.html]: Still not sure how I feel about this particular syntax. 🤷‍♂

File: projects/go-lib/src/lib/components/go-icon/go-icon.component.ts [hunk 5]
```diff
@@ -1,10 +1,25 @@
...
 @Component({
   selector: 'go-icon',
-  templateUrl: 'go-icon.component.html',
-  styleUrls: ['go-icon.component.scss']
+  templateUrl: 'go-icon.component.html'
 })
 export class GoIconComponent {
   @Input() icon: string;
+  @Input() iconModifier: string;
+  @Input() iconClass: string;
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-icon/go-icon.component.ts]: Should we alphabetize these `@Input`s?

File: projects/go-lib/src/lib/components/go-icon/go-icon.component.ts (hunk 2 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-icon/go-icon.component.ts]: Yeah, I don't really love this implementation... but understand the reasoning for it and at the current point in time I can't come up with a better alternative.

File: projects/go-lib/src/lib/components/go-icon/go-icon.component.ts [hunk 6]
```diff
@@ -1,10 +1,25 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 
 @Component({
   selector: 'go-icon',
-  templateUrl: 'go-icon.component.html',
-  styleUrls: ['go-icon.component.scss']
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-icon/go-icon.component.ts]: We need the contents of this sass file. It allows us to properly override the font-size of the icon

Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-icon/go-icon.component.ts]: We adjusted the styling in `gosheets` to fix the issue we were overriding in this file, thus we now do not need it.

**Pull request merged**